ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt

```diff
@@ -0,0 +1,202 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+import net.corda.v5.ledger.common.transaction.Party
+import net.corda.v5.ledger.utxo.Command
+import net.corda.v5.ledger.utxo.StateAndRef
+import net.corda.v5.ledger.utxo.TimeWindow
+import net.corda.v5.ledger.utxo.TransactionState
+import net.corda.v5.ledger.utxo.ContractState
+import java.security.PublicKey
+import java.time.Duration
+import java.time.Instant
+
+/**
+ * Defines a builder for UTXO transactions.
+ *
+ * @property notary The transaction notary.
+ * @property timeWindow The transaction time window.
+ * @property attachments The transaction attachments.
+ * @property commands The transaction commands.
+ * @property inputStateAndRefs The transaction input state and refs.
+ * @property referenceInputStateAndRefs The transaction referenced input state and refs.
+ * @property outputTransactionStates The transaction output states.
+ */
+@DoNotImplement
+@CordaSerializable
+@Suppress("TooManyFunctions")
+interface UtxoTransactionBuilder {
+
+    val notary: Party
+    val timeWindow: TimeWindow
+    val attachments: List<SecureHash>
+    val commands: List<Command>
+    val signatories: Set<PublicKey>
+    val inputStateAndRefs: List<StateAndRef<*>>
+    val referenceInputStateAndRefs: List<StateAndRef<*>>
+    val outputTransactionStates: List<TransactionState<*>>
```

Comment:
Do these need to be public properties? I.e. is there a use case for getting e.g. the notary back from TX builder?

Reply:
I suppose not, although this is something to think about. There are cases where having public access is useful, but perhaps not for everything.